include/seastar/rpc/rpc.hh

```diff
@@ -700,8 +700,11 @@ public:
 protected:
     friend class server;
 
-    virtual rpc_handler* get_handler(uint64_t msg_id) = 0;
-    virtual void put_handler(rpc_handler*) = 0;
+    struct handler_with_holder {
```

Comment:
I think we don't need this wrapper. Just return the rpc_handler to the caller and make _it_ enter the gate and propagate the holder down the stack

Reply:
This was discussed in [this comment](https://github.com/scylladb/seastar/pull/2044#discussion_r1452073859) - this wrap allows for less code repetition and preserve the current code structure.

Reply:
>  ... and preserve the current code structure.

It's not a benefit, but anyway, since everybody else prefers yet-another-wrapper I'm fine too